docs/devnet.md

```diff
@@ -21,9 +21,9 @@
 | NFT Bridge         |    SOL    |                                                           NFTWqJR8YnRVqPDvTJrYuLrQDitTG5AScqbeghi4zSA |                                                                                                                                                                     |
 | Migration Contract |    SOL    |                                                          Ex9bCdVMSfx7EzB3pgSi2R4UHwJAXvTw18rBQm5YQ8gK |                                                                                                                                                                     |
 | Test Wallet        |   Terra   |                                                          terra1x46rqay4d3cssq8gxxvqz8xt6nwlz4td20k38v | Mnemonic: `notice oak worry limit wrap speak medal online prefer cluster roof addict wrist behave treat actual wasp year salad speed social layer crew genius`      |
-| Test CW20          |   Terra   |                                                          terra13nkgqrfymug724h8pprpexqj9h629sa3ncw7sh | Tokens minted to Test Wallet                                                                                                                                        |
-| Bridge Core        |   Terra   |                                                          terra18vd8fpwxzck93qlwghaj6arh4p7c5n896xzem5 |                                                                                                                                                                     |
-| Token Bridge       |   Terra   |                                                          terra10pyejy66429refv3g35g2t7am0was7ya7kz2a4 |                                                                                                                                                                     |
+| Test CW20          |   Terra   |                                      terra1zwv6feuzhy6a9wekh96cd57lsarmqlwxdypdsplw6zhfncqw6ftqynf7kp | Tokens minted to Test Wallet                                                                                                                                        |
+| Bridge Core        |   Terra   |                                      terra14hj2tavq8fpesdwxxcu44rty3hh90vhujrvcmstl4zr3txmfvw9ssrc8au |                                                                                                                                                                     |
+| Token Bridge       |   Terra   |                                      terra1nc5tatafv6eyq7llkr2gv50ff9e22mnf70qgjlv737ktmt4eswrquka9l6 |                                                                                                                                                                     |
```

Comment:
Terra and Terra2 have the same addresses by design?

Reply:
Yeah, the contract address is derived from the `codeId` (incrementally assigned globally) and the `instantiationId` (incrementally assigned per `codeId`). The actual derivation logic is here https://github.com/cosmwasm/wasmd/blob/017f91ccc9affbe030867925ccb11a7ad49133a7/x/wasm/keeper/addresses.go#L36-L41, it involves hashing those plus some constants and padding bytes.

This means that all cosmwasm chains will share the same 32 byte address for these contracts, though the human-readable (Bech32) representation will depend on the first part of the address, the "human readable part". Terra and Terra2 share this part, it's `terra` in both cases, so the human readable address is rendered the same.